tests/ml_tools/test_transformed_module.py

```diff
@@ -42,7 +42,7 @@ def quantum_circuit(n_qubits: int = 2, depth: int = 1) -> QuantumCircuit:
 
 
 def get_qnn(n_qubits: int, depth: int) -> QNN:
-    observable = total_magnetization(n_qubits)
+    observable = hamiltonian_factory(n_qubits, detuning=Z)
```

Comment:
Do we go fo replacing all `total_magnetization` constructors with calls to `hamiltonian_factory` ?

Reply:
yes, i will do it in the MR where i clean up the tests